src/Microsoft.AspNetCore.Mvc.Core/Internal/MvcCoreLoggerExtensions.cs

```diff
@@ -639,7 +645,22 @@ public static IDisposable ActionScope(this ILogger logger, ActionDescriptor acti
 
         public static void ExecutingAction(this ILogger logger, ActionDescriptor action)
         {
-            _actionExecuting(logger, action.DisplayName, null);
+            var routeKeys = action.RouteValues.Keys.ToArray();
+            var routeValues = action.RouteValues.Values.ToArray();
+            string route = "\"";
```

Comment:
Use a stringbuilder here, any time you have strings with `+=` and a loop it's a good idea.